### ISSUE TITLE
AlexNet, ResNet... fail to compile in the onnxmlirczar/onnx-mlir:s390x container

### ISSUE THREAD

**Issue by cfig-xk** [onnx/onnx-mlir#1655]
When compiling AlexNet.onnx, ResNet50.onnx models using `onnx-mlir --EmitLib AlexNet.onnx` in the onnxmlirczar/onnx-mlir:s390x container, I got the following error:
`error: reduction dimension must be the same
error: Failed to scan onnx.MatMul parameters successfully
error: shape inference failed`

Notes:

- I saved the AlexNet in 'tf' format based on python3.6 tensorflow2.4.0 Ubuntu 20.04.4 LTS Linux version 5.15.0-41-generic s390x,  here is the code:
`import tensorflow as tf
import pathlib
import random

from keras.layers import Flatten,BatchNormalization

from keras.layers.convolutional import Conv2D
from keras.layers.merge import add
from keras.models import Sequential

from keras.layers import Input, merge, ZeroPadding2D
from keras.layers import Dense, Activation
from keras.layers.core import Dropout
from keras.layers.convolutional import Convolution2D
from keras.layers.pooling import AveragePooling2D, GlobalAveragePooling2D, MaxPooling2D
from keras.layers.normalization import BatchNormalization

import six
from keras.regularizers import l2
from keras.models import Model

import keras.backend as K
K.set_image_data_format('channels_last')

try:
    from keras import initializations
except ImportError:
    from keras import initializers as initializations

os.environ["CUDA_VISIBLE_DEVICES"]="-1"

class Config(object):
    normal_size = 227
    channles = 3
    classNumber = 1000

class MODEL(object):

    def __init__(self,config):
        self.config = config

    # AlexNet
    def AlexNet(self):
        model = Sequential()
        # input_shape = (64,64, self.config.channles)
        input_shape = (self.config.normal_size, self.config.normal_size, self.config.channles)
        model.add(Convolution2D(96, (11, 11), input_shape=input_shape,strides=(4, 4),  padding='valid',activation='relu', kernel_initializer='random_normal'))
        model.add(MaxPooling2D(pool_size=(3, 3), strides=(2, 2)))#26*26
        model.add(Convolution2D(256, (5, 5), strides=(1, 1), padding='same', activation='relu', kernel_initializer='random_normal'))
        model.add(MaxPooling2D(pool_size=(3, 3), strides=(2, 2)))
        model.add(Convolution2D(384, (3, 3), strides=(1, 1), padding='same', activation='relu', kernel_initializer='random_normal'))
        model.add(Convolution2D(384, (3, 3), strides=(1, 1), padding='same', activation='relu', kernel_initializer='random_normal'))
        model.add(Convolution2D(256, (3, 3), strides=(1, 1), padding='same', activation='relu', kernel_initializer='random_normal'))
        model.add(MaxPooling2D(pool_size=(2, 2), strides=(2, 2)))
        model.add(Flatten())
        # model.add(Dense(4096, activation='relu'))
        model.add(Dense(1024, activation='relu'))
        model.add(Dropout(0.5))
        # model.add(Dense(4096, activation='relu'))
        model.add(Dense(1024, activation='relu'))
        model.add(Dropout(0.5))
        model.add(Dense(self.config.classNumber, activation='softmax'))
        # model.compile(loss="categorical_crossentropy", optimizer="adam", metrics=["accuracy"])
        return model

C = Config()
model = MODEL(C)
alexnet = model.AlexNet()
alexnet.summary()

alexnet.compile(optimizer=tf.keras.optimizers.Adam(lr=0.001),
            loss='sparse_categorical_crossentropy',
            metrics=['accuracy'])

alexnet.save('.\models-s390x\AlexNet', save_format='tf')`

- I convert the AlexNet to onnx based on python3.6 tensorflow2.4.0 Ubuntu 20.04.4 LTS Linux version 5.15.0-41-generic s390x, here is the code:
`import tensorflow as tf
import onnx
import tf2onnx
import os
import argparse

import json
os.environ["CUDA_VISIBLE_DEVICES"]="-1"    

def convert_to_onnx(model, output):
    print('Input model: ' + model)
    tf_model = tf.keras.models.load_model(model)
    # print(tf_model.summary()) 
    onnx_model_proto, _ = tf2onnx.convert.from_keras(tf_model, output_path=output)
    print('Save onnx model to ' + output)

def main():
    parser = argparse.ArgumentParser(description="convert keras model to onnx")
    parser.add_argument('--saved-model', dest='model', help='saved model')
    parser.add_argument('--output', dest='output', help='onnx model')
    args = parser.parse_args()
    convert_to_onnx(args.model, args.output)

if __name__ == "__main__":
    main()
`

**Comment by chentong319** [comment]
Can you provide your .onnx file?

**Comment by cfig-xk** [comment]
> Can you provide your .onnx file?

Thanks for your reply! I have uploaded the onnx files

**Comment by cfig-xk** [comment]
> Can you provide your .onnx file?

Thank you for your reply! The onnx files are too big to be attached. I have uploaded the AlexNet.onnx and ResNet50.onnx to Box, and you can download it through this link:[https://app.box.com/s/wrir7z7uyblb7qb625qrzgj7qxulbjqh](url)

**Comment by cfig-xk** [comment]
> Can you provide your .onnx file?

Thank you for your reply! The onnx files are too big to be attached. I have uploaded the AlexNet.onnx and ResNet50.onnx to oneDrive, and you can download it through this link: [https://1drv.ms/u/s!AgwO8y5CADa4jTT6iVYUOViUnUFT?e=EQmS7K]

**Comment by cfig-xk** [comment]
> Can you provide your .onnx file?
Thank you for your reply! The onnx files are too big to be attached. I have uploaded the AlexNet.onnx and ResNet50.onnx to drive.google.com, and you can download it through this link:
usp=sharing[https://drive.google.com/file/d/1EIN0wignnizyx_RQVwziXvYnSsgFrE26/view?usp=sharing](url)

**Comment by AlexandreEichenberger** [comment]
@cfig-xk is this still an issue with the latest releases?

**Comment by cfig-xk** [comment]
@AlexandreEichenberger I tried compiling the above two models in the latest released container, but got a new error:
`error while loading shared libraries: libChloOps.so.16git: cannot open shared object file: No such file or directory`

**Comment by gongsu832** [comment]
@yaochengji @Connor-XY `libChloOps.so.16git` is part of `third_party/mlir-hlo`.  It's not being installed by `make install`.

**Comment by yaochengji** [comment]
> @yaochengji @Connor-XY `libChloOps.so.16git` is part of `third_party/mlir-hlo`. It's not being installed by `make install`.

A straightforward way to fix this is to add `ChloDialect` to the install targets of `src/Conversion/ONNXToMhlo/CMakeLists.txt`. But I think we need to fix it systematically to avoid further similar bugs.

**Comment by gongsu832** [comment]
Can we get a fix for this relatively soon? I'm about to put in a Jenkins step that actually verifies that our user docker image can successfully compile a simple model. Right now, our user docker image is broken and unusable.

**Comment by yaochengji** [comment]
@gongsu832 sorry for the delay, https://github.com/onnx/onnx-mlir/pull/1829
